README.md

```diff
@@ -379,6 +379,34 @@ Succeeded   : 50
+  - Almost every class has a `to_dict()` method to generate a dictionary representation of an instance of that class. This is used primarily to serialize the objects to JSON between steps in the Nextflow workflow. Ideally, the Python package can deserialize these JSON files into the appropriate set of objects.
+  - In the current version, I distinguish between internal and external tests. This is used to separate tests into two branches in the Nextflow DAG.
+  - The QC suite is intended to represent the set of tests that are relevant to each file type. This approach might need to be re-engineered a bit to accommodate multi-file QC targets though. I'm also using inheritance to accumulate additional QC tests. For example, the OME-TIFF QC suite has OME-specific tests in addition to those inherited from the TIFF QC suite, which in turn inherits from the generic file QC suite (with basic tests like MD5 checksum validation).
+
+### Caveats
+
+- The pipeline currently uses container images on Docker Hub, which might be affected by rate limits.
+- Currently, the Nextflow workflow is set up to run the external command(s) in one process, which means that we are limited to a single container. While we should be able to run multiple commands, if they require different programs, we could either build a composite/mulled containers, refactor the Nextflow workflow to allow for an arbitrary number of steps, or run a WMS (_e.g._ cwltool) within a Nextflow process. Depending on how common this need is, we can decide on which solution is most appropriate. For example, if it's a rare occurrence, then composite/mulled containers would be a reasonable approach.
+- Special care is taken to make sure that the file names are unique in order to avoid name clashes during the two aggregation processes at the end of the workflow. Currently, this involves naming files based on the Synapse ID and the QC test name, but this doesn't generalize well. For example, what would take the place of the Synapse ID in the case of a HTTPS file? Perhaps the unique identifiers can be the index of the data file in the input CSV file (_i.e._ the row number).
+
+### Future Directions
+
+- For now, the workflow is focusing on QC status (_i.e._ `pass`, `fail`, and `skip`). In the future, we can look into generating QC metrics and even build those into the relevant data models so the metrics can be captured as metadata. This may involve leveraging MultiQC.
+- We cannot leverage jobs cached by Nextflow because we cannot rely on intermediate files being stored indefinitely. Hence, we need to build in a separate method for skipping files that have already been QC'ed.
```

Comment:
I have some ideas about this here, but I'd like to hear your thoughts.  We can have a separate ticket to track this in the future.

Reply:
Sounds good!